### PR TITLE
Added py-eprosima-fastdds package

### PR DESCRIPTION
**Pull request by danielecesarini**
This is a python interface for eprosima-fastdds package

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the home page and version sha256.  Adding @adamjstewart and @pradyunsg since they are python package gurus.

**Comment by tldahlgren** [comment]
Also added @manuelakuhn since she is ramping up for the python package as well.

File: var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    which provides publisher-subscriber communications over unreliable transports
+    such as UDP, as defined and maintained by the Object Management Group (OMG) consortium.
+    This is the python interface for eprosima Fast DDS."""
+
+    homepage = "https://www.eprosima.com/"
+    url = "https://github.com/eProsima/Fast-DDS-python/archive/v1.2.2.tar.gz"
+
+    version("1.2.2", sha256="78c53739a66544b8c91d0016560c267e11bd7fdaf727b3bfbffd44ae65c93c62")
+
+    depends_on("python@3.6:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py]: Spack only supports Python 3.7+ so you could probably remove this line

File: var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    depends_on("swig")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py]: Swig is usually only needed at build-time

**Review by adamjstewart — COMMENTED** [review]
Don't know much about CMake so not sure how much more I can help

File: var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py (hunk 1 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py]: I don't think you can totally remove the line because you still need a python dependency. This is a package that inherits from CMakePackage and not PythonPackage (like our usual py-... packages) where python is automatically added as dependency.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py]: Good point. @danielecesarini does the package install any Python libraries in `lib/pythonX.Y/site-packages`? If it does, it should also subclass `PythonExtension`

File: var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    depends_on("eprosima-fastdds")
+    depends_on("openssl")
+
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py:28]: For the tests to even run there is a `py-test` dependency missing
```
    depends_on("py-pytest", type="test")
```

**Review by manuelakuhn — COMMENTED** [review]
According to [CMakeList.txt](https://github.com/eProsima/Fast-DDS-python/blob/main/fastdds_python/CMakeLists.txt#L15) there is minimum version requirement for cmake of `cmake@3.15:`

Also although the installation of the package runs through, the tests with `--test=root` fail with the error
```
E   ModuleNotFoundError: No module named 'fastdds'
```

File: var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py (hunk 1 above)
Comment (danielecesarini) [reply var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py]: #[c28330e](https://github.com/spack/spack/pull/38877/commits/c28330ee921471fc560a16d26c2aceab0ce8ac31)

File: var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py (hunk 2 above)
Comment (danielecesarini) [reply var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py]: #[d6bde44](https://github.com/spack/spack/pull/38877/commits/d6bde446d7046d82ef709063d082075efea72eb5)

File: var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py (hunk 3 above)
Comment (danielecesarini) [reply var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py:28]: #[d6bde44](https://github.com/spack/spack/pull/38877/commits/d6bde446d7046d82ef709063d082075efea72eb5)
#[d8d8894](https://github.com/spack/spack/pull/38877/commits/d8d8894f50609fe6682c1226cc061f41e709dab3)

File: var/spack/repos/builtin/packages/foonathan-memory/package.py [hunk 4]
```diff
@@ -13,4 +13,5 @@ class FoonathanMemory(CMakePackage):
     homepage = "https://memory.foonathan.net/"
     url = "https://github.com/foonathan/memory/archive/v0.7.tar.gz"
 
+    version("0.7-3", sha256="4203d15db22a94a3978eeb1afb59a37d35c57c0f148733f0f1a53a6281cb74dd")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/foonathan-memory/package.py]: This should be in a separate PR since it's unrelated

File: var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    such as UDP, as defined and maintained by the Object Management Group (OMG) consortium.
+    This is the python interface for eprosima Fast DDS."""
+
+    homepage = "https://www.eprosima.com/"
+    url = "https://github.com/eProsima/Fast-DDS-python/archive/v1.2.2.tar.gz"
+
+    version("1.2.2", sha256="78c53739a66544b8c91d0016560c267e11bd7fdaf727b3bfbffd44ae65c93c62")
+
+    depends_on("cmake@3.15:")
+    depends_on("python@3.7:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py]:
```suggestion
    extends("python")
```
This ensures that `PYTHONPATH` gets set to `<prefix>/lib/pythonX.Y/site-packages`

File: var/spack/repos/builtin/packages/foonathan-memory/package.py (hunk 4 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/foonathan-memory/package.py]: Already upgraded, see #38879

File: var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    which provides publisher-subscriber communications over unreliable transports
+    such as UDP, as defined and maintained by the Object Management Group (OMG) consortium.
+    This is the python interface for eprosima Fast DDS."""
+
+    homepage = "https://www.eprosima.com/"
+    url = "https://github.com/eProsima/Fast-DDS-python/archive/v1.2.2.tar.gz"
+
+    version("1.2.2", sha256="78c53739a66544b8c91d0016560c267e11bd7fdaf727b3bfbffd44ae65c93c62")
+
+    depends_on("cmake@3.15:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-eprosima-fastdds/package.py]:
```suggestion
    depends_on("cmake@3.15:", type="build")
```

**Review by adamjstewart — DISMISSED** [review]

**Comment by danielecesarini** [comment]
Fixed GIT history and removed commit on version update of another package

**Review by adamjstewart — APPROVED** [review]
